master/internal/api_config_policies.go

```diff
@@ -74,21 +82,70 @@ func (*apiServer) GetGlobalConfigPolicies(
 }
 
 // Delete workspace task config policies.
-func (*apiServer) DeleteWorkspaceConfigPolicies(
+func (a *apiServer) DeleteWorkspaceConfigPolicies(
 	ctx context.Context, req *apiv1.DeleteWorkspaceConfigPoliciesRequest,
 ) (*apiv1.DeleteWorkspaceConfigPoliciesResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = workspace.AuthZProvider.Get().CanGetWorkspaceID(
+		ctx, *curUser, req.WorkspaceId,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var w model.Workspace
+	err = db.Bun().NewSelect().Model(&w).Where("id = ?", req.WorkspaceId).Scan(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	wksp, err := w.ToProto()
+	err = workspace.AuthZProvider.Get().CanModifyWorkspaceConfigPolicies(ctx, *curUser, wksp)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Hmm, I kinda feel like that adds unnecessary abstraction 🤔 I personally like being able to look at the API handler and clearly see exactly what it's doing unless the code is particularly long and repetitive.
Also, in this case, part of the db queries are for request validation and the other half (the call to `DeleteConfigPolicies` is for fulfilling the request.
WDYT about leaving this as is?

Reply:
Why not use `a.GetWorkspaceByID` instead of a raw bun query? I really think raw bun queries should not be in the API function.

I also think that the primary purpose of getting the workspace is to validate the user permissions for that workspace. I don't think we need to check that the workspace exists, in this case.

Reply:
After our convo about deciding to error out in the case of an invalid workspace ID, I substituted the bun query with `a.GetWorkspaceByID`

Reply:
Perfect, thank you!